providers/os/resources/firewalld.go

```diff
@@ -0,0 +1,1309 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package resources
+
+import (
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/spf13/afero"
+	"go.mondoo.com/cnquery/v12/llx"
+	"go.mondoo.com/cnquery/v12/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/v12/providers/os/connection/shared"
+	"go.mondoo.com/cnquery/v12/providers/os/resources/parsers"
+	"go.mondoo.com/cnquery/v12/types"
+)
+
+const defaultFirewalldConfig = "/etc/firewalld/firewalld.conf"
+
+var firewalldZoneSearchPaths = []string{
+	"/run/firewalld/zones",
+	"/etc/firewalld/zones",
+	"/usr/lib/firewalld/zones",
+	"/usr/share/firewalld/zones",
+}
+
+type mqlFirewalldInternal struct {
+	lock      sync.Mutex
+	loaded    bool
+	loadError error
+}
+
+type firewalldData struct {
+	DefaultZone string
+	ActiveZones []string
+	Zones       []firewalldZoneData
+}
+
+type firewalldZoneData struct {
+	Zone  parsedFirewalldZone
+	Rules []parsedFirewalldRule
+}
+
+type firewalldZoneFile struct {
+	Name    string
+	Path    string
+	Content string
+}
+
+type firewalldZoneXML struct {
```

Comment:
the structs and the parser should be extracted into a separate file as its not really readable this way. Since firewalld parsing is more complex, it should be splitted from MQL resource handling

Reply:
when we extract the xml parser into its own package, we should also have tested it there.